### ISSUE TITLE
Question regarding latent embedding

### ISSUE THREAD

**Issue by HannahHaensen** [zubair-irshad/CenterSnap#11]
Is the latent embedding required?

https://github.com/zubair-irshad/CenterSnap/blob/c2afd120428b0a07c88894da23311995b72bbbfd/prepare_data/generate_data_nocs.py#L202

how to obtain this from a new dataset? do i have to train NOCS?

**Comment by HannahHaensen** [comment]
thank you for the detailed answer! I will try it

**Comment by yuanzhen2020** [comment]
Great work! May i ask the GT latent embedding code for training is generated through the CAD model of training set? And is it possible to generate latent embedding code from partial point cloud (obtained by masked depth).

**Comment by zubair-irshad** [comment]
1. Yes, that's correct. We generate the GT latent embedding code by training an auto-encoder using the CAD models available in the training set (Please also see above response for a detailed answer). 
2. Unfortunately it is not possible to use partial pointclouds i.e. masked depth maps directly since it would not give the proper size of the objects which is crucial for obtaining the accurate size information needed for absolute pose information and eventual accurate reconstruction. Please see [this](https://github.com/zubair-irshad/CenterSnap/blob/master/inference/inference_real.py#L95) and [this](https://github.com/zubair-irshad/CenterSnap/blob/189021973e2f5997dd2df96b3d079161433d6e3a/utils/transform_utils.py#L55-L59) on how we obtain sizes and rotated bounding boxes from canonical bounding boxes. 
Note that we use these CAD models and GT latent embedding as a strong prior so we can generalize to unseen images (i.e. we do not require any CAD models during testing, we only require a single RGB-D image during testing) and hence the more accurate the prior is learnt, the better the generalization performance would be.

Hope it helps!

**Comment by HannahHaensen** [comment]
I have another question regarding this pretrainin in the paper it says you use the shapenet CAD models but here the CAD models from NOCS are used or do I get the README wrong?

**Comment by zubair-irshad** [comment]
@HannahHaensen, The CAD models used by NOCS are exactly similar to ShapeNet. The difference is, NOCS trains on only 6 Shapenet Categories and selects a subset of ShapeNet models for which they render the synthetic images on table top scenes i.e. their train set. For the paper, we only train on the subset of ShapeNet models (6 categories only) used by NOCS but we have also trained our auto-encoder on all shapenet models (unfortunately we cannot release the pre-trained checkpoints for that). Note that if you wish to, you can train the auto-encoder on all shapenet categories using the same [train script](https://github.com/zubair-irshad/CenterSnap/blob/master/external/shape_pretraining/train_ae.py). Hope this helps!
